netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -467,6 +469,25 @@ private void gracefulClose(ChannelHandlerContext ctx, GracefulCloseCommand msg,
     close(ctx, promise);
   }
 
+  private void forcefulClose(final ChannelHandlerContext ctx, final ForcefulCloseCommand msg,
+      ChannelPromise promise) throws Exception {
+    lifecycleManager.notifyShutdown(
+        Status.UNAVAILABLE.withDescription("Channel requested transport to shut down"));
```

Comment:
Maybe attach msg.getStatus().asException() as the cause of this UNAVAILABLE status, and also use it in transportReportStatus()? This keeps the most information and makes the event chain more obvious.

I don't have an opinion between UNAVAILABLE or CANCELLED for transportReportStatus().